FirebaseAppCheck/Sources/DeviceCheckProvider/FIRDeviceCheckProvider.m

```diff
@@ -16,10 +16,12 @@
 
 #import "FirebaseAppCheck/Sources/Public/FirebaseAppCheck/FIRAppCheckAvailability.h"
 
-#if FIR_DEVICE_CHECK_SUPPORTED_TARGETS
-
 #import <Foundation/Foundation.h>
 
+#if __has_include(<DeviceCheck/DeviceCheck.h>)
+#import <DeviceCheck/DeviceCheck.h>
+#endif
```

Comment:
None of the code below makes any sense if `DeviceCheck` is unavailable. I wonder why you would prefer to still keep include it?

Reply:
This is entirely so it shows up in Xcode if a user tries to invoke any of these APIs with a helpful message saying "this is unavailable on the platform you're targeting" instead of saying "No known method for selector".

Reply:
I probably don't have enough context here. Could you please give an example of what error we are trying to avoid and what will be the replacement. I would like to understand better if the unnecessary binary size increase is a fair trade of here.

Reply:
Here's an example where a user was trying to find a method in Auth: https://github.com/firebase/firebase-ios-sdk/issues/7127#issuecomment-740850842

I'll see if I can expose the more helpful method without increasing binary size on platforms where the DeviceCheck classes are unused.

Reply:
I wasn't able to find a way to keep the more helpful diagnostics without increasing the code size. If this module is built statically and linked without the `-ObjC` flag, the linker would be able to remove the two unused object files on watchOS, but that seems unlikely for most app targets. For a dynamically-built module the symbols would always be included in the final app bundle (from what I understand).

I'm fine with reverting these changes except the documentation comments, let me know what you think.

Reply:
I don't feel too strongly about it, but excluding the code on completely unsupported platform (watchOS) looks a bit cleaner to me. In this specific case it should not create too much confusion because the underlying Apple API also is completely unavailable.

Reply:
Done.